includes/Core/Modules/Module.php

```diff
@@ -339,9 +341,14 @@ public function is_connected() {
 			} catch ( Exception $e ) {
 				// Set every result of this batch to the exception.
 				foreach ( $results as $key => $definition_key ) {
-					$datapoint_service = ! empty( $datapoint_definitions[ $definition_key ] ) ? $datapoint_definitions[ $definition_key ]['service'] : null;
-					if ( is_string( $definition_key ) && $service_identifier === $datapoint_service ) {
-						$results[ $key ] = $this->exception_to_error( $e, explode( ':', $definition_key, 2 )[1] );
+					if ( ! is_wp_error( $definition_key ) ) {
```

Comment:
Let's change this to (below) instead
```js
if ( is_wp_error( $definition_key ) ) {
	continue;
}
```
This will avoid the need to further nest the code here and make it both a smaller change and easier to read 👍

Reply:
Updated.